pyroute2/netlink/nlsocket.py

```diff
@@ -175,6 +178,11 @@ def parse(self, data, seq=None, callback=None):
 
             try:
                 msg.decode()
+                # TODO: not the good method to do that
```

Comment:
Why don't you think it's a proper place to parse an error message?

Reply:
In my opinion, this should be in msg.decode() itself, if possible. It looks weird for me to call msg.decode() twice

And it's not only about error parsing. I hardcoded the nlmsgerr class here, but we can receive some "cookies" with relevant data when extack is enabled